### PR TITLE
acl: oss plumbing to support auth method namespace rules in enterprise

### PR DESCRIPTION
**Pull request by rboyer**
This includes website docs updates.

### CODE REVIEW

File: website/pages/docs/acl/auth-methods/kubernetes.mdx [hunk 1]
```diff
@@ -133,11 +136,11 @@ roleRef:
...
 use in binding rule selectors and bind name interpolation.
 
-| Attributes                 | Supported Selector Operations | Can be Interpolated |
-| -------------------------- | ----------------------------- | ------------------- |
-| `serviceaccount.namespace` | Equal, Not Equal              | yes                 |
-| `serviceaccount.name`      | Equal, Not Equal              | yes                 |
-| `serviceaccount.uid`       | Equal, Not Equal              | yes                 |
+| Attributes                 | Supported Selector Operations                      | Can be Interpolated |
+| -------------------------- | -------------------------------------------------- | ------------------- |
+| `serviceaccount.namespace` | Equal, Not Equal, In, Not In, Matches, Not Matches | yes                 |
```
Comment (rboyer) [inline website/pages/docs/acl/auth-methods/kubernetes.mdx:141]: This is an unrelated correction.

File: website/pages/api-docs/acl/auth-methods.mdx [hunk 2]
```diff
@@ -70,6 +70,31 @@ The table below shows this endpoint's support for
   If not provided at all, the namespace will be inherited from the request's ACL
   token or will default to the `default` namespace. Added in Consul 1.7.0.
 
+- `NamespaceRules` `(array<NamespaceRule>)` <EnterpriseAlert inline /> - A set
+  of rules that can control which namespace tokens created via this auth method
+  will be created within. Note that assigning namespaces via rules requires the
```
Comment (s-christoff) [inline website/pages/api-docs/acl/auth-methods.mdx:75]: Should we sub "will be created within" to something like "will be associated with"?

File: website/pages/docs/acl/auth-methods/kubernetes.mdx [hunk 3]
```diff
@@ -42,17 +42,20 @@ parameters are required to properly configure an auth method of type
   namespace instead of creating tokens in the auth methods own namespace. Note
   that mapping namespaces requires the auth method to reside within the
   `default` namespace.
+  Deprecated in Consul 1.8.0 in favor of [namespace rules](/api/acl/auth-methods#namespacerules).
 
 - `ConsulNamespacePrefix` `(string: <optional>)` <EnterpriseAlert inline /> - When
   `MapNamespaces` is enabled, this value will be prefixed to the Kubernetes
   namespace to determine the Consul namespace to create the new token within.
+  Deprecated in Consul 1.8.0 in favor of [namespace rules](/api/acl/auth-methods#namespacerules).
```
Comment (s-christoff) [inline website/pages/docs/acl/auth-methods/kubernetes.mdx:50]: I think it may be better to add this deprecated sentence at the beginning

File: website/pages/api-docs/acl/auth-methods.mdx (hunk 2 above)
Comment (rboyer) [reply website/pages/api-docs/acl/auth-methods.mdx:75]: Tokens are stored within a namespace. If you delete a namespace everything inside it is erased.

Comment (s-christoff) [reply website/pages/api-docs/acl/auth-methods.mdx:75]: Okay, I see. Intentional wording is worded well 👍
